packages/functions/src/scanner.ts

```diff
@@ -0,0 +1,125 @@
+import { Config } from 'sst/node/config'
+import { Table } from 'sst/node/table'
+import { Queue } from 'sst/node/queue'
+import { Context } from 'aws-lambda'
+import { unmarshall } from '@aws-sdk/util-dynamodb'
+import { DynamoDBClient, ScanCommand, ScanCommandOutput } from '@aws-sdk/client-dynamodb'
+import { SQSClient, SendMessageCommand } from '@aws-sdk/client-sqs'
+import { LambdaClient, InvokeCommand } from '@aws-sdk/client-lambda'
+import { SSMClient, PutParameterCommand, GetParameterCommand, ParameterNotFound } from '@aws-sdk/client-ssm'
+
+const SCAN_BATCH_SIZE = 500 // max sqs msg is 256KB. each record is ~350bytes, 350 * 500 = 175KB
+const MIN_REMAINING_TIME_MS = 10_000 // threshold at which we reinvoke the lambda
+
+/**
+ * Lambda to scan an entire DynamoDB table, sending batches of records to an SQS Queue.
+ * 
+ * Stores it's progress in SSM parameter store so it can resume.
+ * 
+ * Invokes itself again when it's remaining execution time is less than `MIN_REMAINING_TIME_MS`.
+ * 
+ * Invoke it directly with 
+ *   aws lambda invoke --function-name <name> --invocation-type Event
+ * 
+ * Pass { TotalSegments: number, Segment: number} to control the table scan partition
+ * see: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Scan.html#Scan.ParallelScan
+ */
+export async function handler(event: any, context: Context) {
+  console.log(`invoked`, event)
+  const TotalSegments = event.TotalSegments ?? 1
+  const Segment = event.Segment ?? 0
+
+  // set via the `bind` config in the sst stack
+  const tableName = Table.srcTable.tableName
+  const queueUrl = Queue.batchQueue.queueUrl
```

Comment:
Are these types working? From my experience with bind config we need to set them, something along the lines of

```ts
declare module 'sst/node/table' {
  export interface TableResources {
    'piece-store': {
      tableName: string;
    };
    'aggregate-store': {
      tableName: string;
    };
    'inclusion-store': {
      tableName: string;
    };
  }
}
```

Reply:
i ran `npx sst types` and it made them exist for me! I didn't dig any further as it seemed to work. I get autocomplete on the things in the lambda impl

Reply:
that's quite nice! I assume if you use typescript project it can get those. Thanks for the tip ❤️